packages/@aws-cdk/aws-certificatemanager/lib/dns_validated_certificate_handler/index.js

```diff
@@ -0,0 +1,196 @@
+ *
+ * @param {string} arn The certificate ARN
+ */
+const deleteCertificate = async function (arn) {
+  const acm = new aws.ACM()
+
+  console.log(`Deleting certificate ${arn}`)
+
+  await acm.deleteCertificate({
+    CertificateArn: arn
+  }).promise()
+}
+
+/**
+ * Main handler, invoked by Lambda
+ */
+exports.certificateRequestHandler = async function (event, context, callback) {
+  var responseData = {}
+  var physicalResourceId
+  var certificateArn
+
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        certificateArn = await requestCertificate(event.RequestId, event.ResourceProperties.DomainName, event.ResourceProperties.HostedZoneId)
+        responseData.Arn = physicalResourceId = certificateArn
+        break
+      case 'Delete':
+        physicalResourceId = event.PhysicalResourceId
+        // If the resource didn't create correctly, the physical resource ID won't be the
+        // certificate ARN, so don't try to delete it in that case.
+        if (physicalResourceId.startsWith('arn:')) {
+          await deleteCertificate(physicalResourceId)
+        }
+        break
+      case 'Update':
+        await deleteCertificate(event.PhysicalResourceId)
+        certificateArn = await requestCertificate(event.RequestId, event.ResourceProperties.DomainName, event.ResourceProperties.HostedZoneId)
+        responseData.Arn = physicalResourceId = certificateArn
+        break
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`)
+    }
+
+    console.log(`Uploading SUCCESS response to S3...`)
+
+    await request({
+      uri: event.ResponseURL,
+      method: 'PUT',
+      body: {
+        Status: 'SUCCESS',
+        PhysicalResourceId: physicalResourceId,
+        StackId: event.StackId,
+        RequestId: event.RequestId,
+        LogicalResourceId: event.LogicalResourceId,
+        Data: responseData
+      },
+      headers: {
+        // This is required to ensure the presigned S3 URL signature matches
+        'Content-Type': ''
+      },
+      json: true,
+      followAllRedirects: true,
+      followOriginalHttpMethod: true
+    })
+
+    console.log('Done.')
+  } catch (err) {
+    console.log(`Caught error ${err}. Uploading FAILED message to S3.`)
+    try {
+      await request({
+        uri: event.ResponseURL,
+        method: 'PUT',
+        body: {
+          Status: 'FAILED',
+          Reason: err.message,
+          PhysicalResourceId: physicalResourceId,
+          StackId: event.StackId,
+          RequestId: event.RequestId,
+          LogicalResourceId: event.LogicalResourceId
+        },
+        headers: {
+          // This is required to ensure the presigned S3 URL signature matches
+          'Content-Type': ''
+        },
+        json: true,
+        followAllRedirects: true,
+        followOriginalHttpMethod: true
+      })
+    } catch (err) {
+      if (callback) {
+        callback(err)
+      }
+    }
+  }
+
+  // Success
+  if (callback) {
```

Comment:
Fairly sure this is unnecessary if the handler func is `async`.

Reply:
I'm sure you're right, but it's useful in a test harness

Reply:
Not sure I understand. Surely in the test harness you could also just do an `await` on the handler's promise?

Speaking of which, could you see your way to adding some unit tests?

Reply:
Eliminated the `callback` parameter -- indeed, it's not used in the ES6 async function model. 

I'll dig into adding unit tests.  Stay tuned.

Reply:
You're right, the `callback` argument isn't needed if you use the ES6 `async function` style handler signature.  Removed it.